### PR TITLE
Add validator service

### PR DESCRIPTION
**Pull request by GevinHasmitha**
When there are errors in the fhir resource;
- Extracted the detailedErrors array from the fhir error returned by the fhir validator.
- Created FHIRIssueDetail instances for each detailed error in the detailedErrors array.
- Using the FHIRIssueDetail instances, created an operationOutcome which is returned.

When fhir resource has errors, returns an http response containing an operationOutcome which has issues for each detailed error.
When validation is successful, returns an http response containing an operationOutcome  with" validation successful" message.

### CODE REVIEW

**Comment by sameeragunarathne** [comment]
Add README file describing what this service is, installation and how to setup etc. Refer other prebuilt services

File: miscellaneous/validator-service/service.bal [hunk 1]
```diff
@@ -0,0 +1,114 @@
+import ballerina/http;
```
Comment (sameeragunarathne) [inline miscellaneous/validator-service/service.bal:1]: Missing licence headers

File: miscellaneous/validator-service/service.bal [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+    (r4:CodeableConcept)? details = detail.details;
+    if details != () {
+        issueBBE.details = details;
+    }
+
+    string? diagnostic = detail.diagnostic;
+    if diagnostic != () {
+        issueBBE.diagnostics = string `${diagnostic}`;
+    } else {
+        issueBBE.diagnostics = "";
```
Comment (sameeragunarathne) [inline miscellaneous/validator-service/service.bal:32]: no need to set a empty string?

Comment (GevinHasmitha) [reply miscellaneous/validator-service/service.bal:32]: Ack

File: miscellaneous/validator-service/service.bal (hunk 1 above)
Comment (GevinHasmitha) [reply miscellaneous/validator-service/service.bal:1]: Ack

File: miscellaneous/validator-service/.devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (sameeragunarathne) [inline miscellaneous/validator-service/.devcontainer.json:1]: No need of this file. let's gitignore it and remove from here

File: miscellaneous/validator-service/service_utils.bal [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+# FHIR validation error details record.
+#
+# + issues - FHIR issues
+public type FHIRValidationErrorDetail record {
+    *r4:FHIRErrorDetail;
+    [validator:FHIRValidationIssueDetail, validator:FHIRValidationIssueDetail...] issues;
+};
+
+isolated function issueDetailToOperationOutcomeIssue(r4:FHIRIssueDetail detail) returns r4:OperationOutcomeIssue {
```
Comment (sameeragunarathne) [inline miscellaneous/validator-service/service_utils.bal:24]: Add bal doc.

File: miscellaneous/validator-service/service.bal [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+
+import ballerina/http;
+import ballerina/log;
+import ballerinax/health.fhir.r4;
+import ballerinax/health.fhir.r4.validator;
+
+configurable string[] CORS_ALLOWED_ORIGINS = ?;
```
Comment (sameeragunarathne) [inline miscellaneous/validator-service/service.bal:18]: Do we need to have the cors config in order the service to be consumed?

File: miscellaneous/validator-service/.devcontainer.json (hunk 3 above)
Comment (GevinHasmitha) [reply miscellaneous/validator-service/.devcontainer.json:1]: Ack

File: miscellaneous/validator-service/service.bal (hunk 5 above)
Comment (GevinHasmitha) [reply miscellaneous/validator-service/service.bal:18]: Yes as this service is for react, cors config is compulsory.

File: miscellaneous/validator-service/service_utils.bal (hunk 4 above)
Comment (GevinHasmitha) [reply miscellaneous/validator-service/service_utils.bal:24]: Ack

**Review by sameeragunarathne — APPROVED** [review]

**Pull request merged**